compiler/lib/driver.ml

```diff
@@ -157,38 +155,20 @@ let rec loop max name round i (p : 'a) : 'a =
     p')
   else loop max name round (i + 1) p'
 
+let round : 'a -> 'a =
+  print +> tailcall +> phi +> flow +> specialize +> eval +> inline +> deadcode
+
 (* o1 *)
 
-let o1 : 'a -> 'a =
-  print
-  +> tailcall
-  +> flow
-  +> specialize
-  +> eval
-  +> inline (* inlining may reveal new tailcall opt *)
-  +> deadcode
-  +> tailcall
-  +> phi
-  +> flow
-  +> specialize
-  +> eval
-  +> inline
-  +> deadcode
-  +> print
-  +> flow
-  +> specialize
-  +> eval
-  +> inline
-  +> deadcode
-  +> phi
+let o1 = loop 2 "round" round 1 +> phi +> flow +> specialize +> eval +> print
```

Comment:
How did you decide to add the `phi +> flow +> specialize +> eval` ?

Reply:
These are just the first passes in `round`. It does not really make sense to end with an inlining pass, since the principal interest of this pass is to allow more simplifications afterward. I have omitted the tailcall pass since there is not much opportunity for more tail call optimizations after two rounds of optimizations.